### PR TITLE
Support new helm chart structure in validation

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/pull/48249

### CODE REVIEW

**Review by jwendell — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Comment by ericvn** [comment]
```
2023-12-20T21:24:58.267456Z	info	Check failed: check HelmOperatorManifest failed: invalid path: expected map or string, got <nil> for defaults. in [defaults. tag]
2023-12-20T21:24:58.267460Z	info	Check failed: check HelmChartVersions failed: cni: invalid path: expected map or string, got <nil> for defaults.global in [defaults.global tag]
2023-12-20T21:24:58.267463Z	info	Check failed: check HelmVersionsIstio failed: invalid path: expected map or string, got <nil> for defaults.global in [defaults.global tag]
```

**Comment by howardjohn** [comment]
@ericvn the two PRs are co-dependant on each other, I will force merge this one once the istio/istio PR is approved.

**Pull request merged**
